src/shared/image_builder_lib.sh

```diff
@@ -0,0 +1,299 @@
+  fi
+
+  echo -n "${additional_tags[@]}"
+}
+
+cicd_tools::image_builder::_array_empty() {
+  local arr=("$1")
+
+  [[ "${#arr[@]}" -eq 1 && -z "${arr[0]}" ]]
+}
+
+cicd_tools::image_builder::get_labels() {
+
+  declare -a labels=("${CICD_TOOLS_IMAGE_BUILDER_LABELS[@]:-${LABELS[@]}}")
+
+  if cicd_tools::image_builder::_array_empty "${labels[@]}"; then
+  labels=()
+  fi
+
+  if cicd_tools::image_builder::is_change_request_context; then
+  labels+=("$(cicd_tools::image_builder::_get_expiry_label)")
+  fi
+
+  echo -n "${labels[@]}"
+}
+
+cicd_tools::image_builder::_get_expiry_label() {
+  echo "quay.expires-after=${CICD_TOOLS_IMAGE_BUILDER_QUAY_EXPIRE_TIME}"
+}
+
+cicd_tools::image_builder::get_build_args() {
+
+  declare -a build_args=("${CICD_TOOLS_IMAGE_BUILDER_BUILD_ARGS[@]:-${BUILD_ARGS[@]}}")
+
+  if cicd_tools::image_builder::_array_empty "${build_args[@]}"; then
+  build_args=()
+  fi
+
+  echo -n "${build_args[@]}"
+}
+
+cicd_tools::image_builder::tag() {
+
+  local source_image target_tag image_name
+  image_name="$(cicd_tools::image_builder::_get_image_name)" || return 1
+  source_image="$(cicd_tools::image_builder::get_full_image_name)" || return 1
+
+  for target_tag in $(cicd_tools::image_builder::get_additional_tags); do
+    if ! cicd_tools::container::cmd tag "$source_image" "${image_name}:${target_tag}"; then
+        cicd_tools::err "Error tagging '$source_image' as '${image_name}:${target_tag}'"
+        return 1
+    fi
+  done
+}
+
+cicd_tools::image_builder::push() {
+
+  local image_name image_tag
+  image_name="$(cicd_tools::image_builder::_get_image_name)" || return 1
+  image_tag=$(cicd_tools::image_builder::get_image_tag) || return 1
+
+  image_tags=("$image_tag")
+
+  if ! cicd_tools::image_builder::is_change_request_context; then
+  for additional_tag in $(cicd_tools::image_builder::get_additional_tags); do
+    image_tags+=("${additional_tag}")
+  done
+  fi
+
+  for tag in "${image_tags[@]}"; do
+  if ! cicd_tools::container::cmd push "${image_name}:${tag}"; then
+    cicd_tools::err "Error pushing image: '${image_name}:${tag}'"
+    return 1
+  fi
+  done
+}
+
+cicd_tools::image_builder::get_full_image_name() {
+
+  local image_name image_tag
+  image_name="$(cicd_tools::image_builder::_get_image_name)" || return 1
+  image_tag=$(cicd_tools::image_builder::get_image_tag) || return 1
+
+  echo -n "${image_name}:${image_tag}"
+}
+
+
+cicd_tools::image_builder::_image_builder_setup() {
+
+  if ! cicd_tools::image_builder::_try_log_in_to_image_registries; then
+    cicd_tools::err "Error trying to log into the image registries!"
+    return 1
+  fi
+}
+
+cicd_tools::image_builder::_try_log_in_to_image_registries() {
+
+  if cicd_tools::image_builder::_quay_credentials_found; then
+  if ! cicd_tools::image_builder::_log_in_to_quay_registry; then
+      cicd_tools::err "Error logging in to Quay.io!"
+      return 1
+  fi
+  fi
+
+  if cicd_tools::image_builder::_redhat_registry_credentials_found; then
```

Comment:
Same question here

Reply:
should be addressed based on previous comment's discussion. Resolving this now.